src/definitions/modules/modal.js

```diff
@@ -1049,7 +1049,7 @@
                             $input = $inputs.first();
                         }
                         if ($input.length > 0) {
-                            $input.trigger('focus');
+                            $input[0].focus();
```

Comment:
I have tested this fix and it fixes https://github.com/fomantic/Fomantic-UI/pull/2575#issuecomment-1399945576

I tried to debug why jQuery focus is not working and noticed this line is called many times. Is that really needed? Maybe that is the source of the issue, as the line is called once even before close (when you click on the dimmer).

Reply:
@mvorisek Thanks a lot for that hint! I reverted the change back to `.trigger('focus')` and disabked the attribute observing while the modal gets shown which now makes sure the autofocus does not trigger multiple times anymore which in turn fixes your issue without the need for the native focus() event

Reply:
Any idea why the jQuery focus was not working? If the problem was frequency, frequent changes can be observed even after the modal is shown.

If I am not mistaken the modal is focused each time a change is observed (when there is no input already focus), but the https://github.com/fomantic/Fomantic-UI/pull/2575#issuecomment-1397852452 repro has no input, and main modal div is focused instead. Can the div focus be detected? If yes, just do nothing if it is in focus already.

Reply:
> Any idea why the jQuery focus was not working? If the problem was frequency, frequent changes can be observed even after the modal is shown.

no idea (i tried to understand jquery logic but did not spent that much time into it tbh) 
So, i can of course still/again use the native .focus() in addition to the observerattributes logic to be 100% sure.. Should i change the code again? opinion?
The issue was only happening when the modal shown and only when the modal did not contain any input.

> 
> If I am not mistaken the modal is focused each time a change is observed (when there is no input already focus), but the [#2575 (comment)](https://github.com/fomantic/Fomantic-UI/pull/2575#issuecomment-1397852452) repro has no input, and main modal div is focused instead. Can the div focus be detected? If yes, just do nothing if it is in focus already.

I know only remove the tabindex from the div by [022381d](https://github.com/fomantic/Fomantic-UI/pull/2674/commits/022381d75163cbbc5b2db65d358816a0d7357741) when there is not input found inside, so in case it was focused before, it doesnt get refocused as `.filter(':focus')` fetches it already

Also fixed a stupid check for class changes (which _always_ triggered the refreshinput before)

Reply:
> So, i can of course still/again use the native .focus() in addition to the observerattributes logic to be 100% sure.. Should i change the code again? opinion?

no, I never like changes without understanding the problem, better to keep jQuery and if the problem will become more problematic later, let's dig into it again

> I now only remove the tabindex from the div by [022381d](https://github.com/fomantic/Fomantic-UI/pull/2674/commits/022381d75163cbbc5b2db65d358816a0d7357741) when there is no input found inside, so in case it was focused before, it doesnt get refocused as .filter(':focus') fetches it already

is `observeAttributes = false` still needed? If not, I prefer to remove it to keep code simple as possible.

Reply:
> is `observeAttributes = false` still needed? If not, I prefer to remove it to keep code simple as possible.

Yes, it is, because the attribute observing for classnames will otherwise trigger the inputrefresh many times while showing/hiding the modal (and if the modal contains inputs).

I thought of completely disconnecting the observer while showing/hiding but that probably breaks projects wich are handling the modal content via ajax and possibly dont trigger the focusing at all.. to risky, so i would leave the code as it is now.